backoffice/backoffice/workflows/api/views.py

```diff
@@ -127,6 +129,65 @@ def resolve(self, request, pk=None):
                 ResolutionDags[serializer.validated_data["value"]].label, pk, extra_data
             )
 
+    @action(detail=True, methods=["post"])
+    def restart(self, request, pk=None):
+        params = request.data.get("params")
+        restart_current_task = request.data.get("restart_current_task", False)
+
+        workflow = Workflow.objects.get(id=pk)
+
+        data = {"dry_run": False, "dag_run_id": pk, "reset_dag_runs": True}
+
+        executed_dags_for_workflow = {}
+        # find dags that were executed
+        for dag_id in AUTHOR_DAGS[workflow.workflow_type]:
+            response = requests.get(
+                f"{airflow_utils.AIRFLOW_BASE_URL}/api/v1/dags/{dag_id}/dagRuns/{pk}",
+                json=data,
+                headers=airflow_utils.AIRFLOW_HEADERS,
+            )
+            if response.status_code == status.HTTP_200_OK:
+                executed_dags_for_workflow[dag_id] = response.content
+
+        #  assumes current task is one of the failed tasks
+        if restart_current_task:
```

Comment:
here the workflows are already restarted right? so there is not much of a point for restarting one task. Do I miss something? Instead we should either restart all or restart a specific task

Reply:
in here the worfklows haven not yet been restarted. the section of code before it is just checking what dags ran for this workflow. moving this section 141-150 below to make it more clear